crates/uv/src/commands/toolchain/list.rs

```diff
@@ -0,0 +1,70 @@
+use std::fmt::Write;
+use std::ops::Deref;
+
+use anyhow::Result;
+use itertools::Itertools;
+
+use uv_cache::Cache;
+use uv_configuration::PreviewMode;
+use uv_toolchain::downloads::PythonDownloadRequest;
+use uv_toolchain::managed::InstalledToolchains;
+use uv_warnings::warn_user;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+use crate::settings::ToolchainListIncludes;
+
+/// List available toolchains.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn list(
+    includes: ToolchainListIncludes,
+    preview: PreviewMode,
+    _cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user!("`uv toolchain list` is experimental and may change without warning.");
+    }
+
+    let downloads = match includes {
+        ToolchainListIncludes::All => {
+            let request = PythonDownloadRequest::default();
+            request.iter_downloads().collect()
+        }
+        ToolchainListIncludes::Installed => Vec::new(),
+        ToolchainListIncludes::Default => {
+            let request = PythonDownloadRequest::from_env()?;
+            request.iter_downloads().collect()
+        }
+    };
```

Comment:
This could be refactored to avoid an allocation:
```rust
    let download_request = match includes {
        ToolchainListIncludes::All => Some(PythonDownloadRequest::default()),
        ToolchainListIncludes::Installed => None,
        ToolchainListIncludes::Default => Some(PythonDownloadRequest::from_env()?)
    };

    let downloads = download_request
        .as_ref()
        .map(|request| request.iter_downloads())
        .into_iter()
        .flatten();
```

Reply:
Thanks!